### PR TITLE
Add lessonPlan to allowed values [SA-15583] develop

### PR DESCRIPTION
**Pull request by tomcdonnell**
Updated docs:
image

Actual responses:
image
image

### CODE REVIEW

**Comment by stevechan** [comment]
the docs don't mention that `participants` are nullable... @JustinAClarke @armatronic  is there supposed to be some kind of audit, or how much thoroughness was supposed to be here? idk, but I'd presume that we'd want our docs to match reality.

**Comment by tomcdonnell** [comment]
> the docs don't mention that `participants` are nullable...

I have set `participants` to be nullable in the docs, but the description 'array of objects' is the same as before.  As far as I can tell, this is due to a problem with the OpenAPI site generator.  By contrast, the auto-generated description 'object' changes to 'object or null' when `nullable` is set to true.  See also folder->yearLevel as another example of a nullable array of objects that is described only as 'array of object' in the generated docs.

**Comment by jleckie** [comment]
We will need a master branch of this change aswell.. As lesson plans went out with the 23.0 release.. So we will likely need to tag a docs update to 23.0.1

File: openapi/components/schemas/assessment-item.yaml [hunk 1]
```diff
@@ -50,10 +51,14 @@ properties:
               description: The name of the year level.
   commonAssessment:
     type: boolean
+    nullable: true
+    description: Null if assessmentType is 'lessonPlan'.
```
Comment (jleckie) [inline openapi/components/schemas/assessment-item.yaml:55]: I am pretty sure more than just lesson plans will have null for common assessment...  In fact the only object that can have a commonAssessment flag is "Due Work"

File: openapi/components/schemas/assessment-item.yaml [hunk 2]
```diff
@@ -65,11 +70,13 @@ properties:
         example: Formative Assessment
   weight:
     type: integer
+    nullable: true
     example: 30
     description: |
       The value of how much this assessment's outcome contributes to the final overall grade -
       a value of 0 indicates this assessment has no impact on the overall grade.
       Generally, this is a number between 0 and 100.
+      Null if assessmentType is 'lessonPlan'.
```
Comment (jleckie) [inline openapi/components/schemas/assessment-item.yaml:79]: This is also null if it is a 'task'..

**Pull request closed without merging**